<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HoleFilling</class>
 <widget class="QDockWidget" name="HoleFilling">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>815</width>
    <height>973</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hole Filling</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Action:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="action_combo_box">
        <property name="currentIndex">
         <number>2</number>
        </property>
        <item>
         <property name="text">
          <string>Triangulate</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Triangulate + Refine</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Triangulate + Refine + Fair</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Hole Filling Parameters</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QCheckBox" name="Use_delaunay_triangulation_check_box">
         <property name="text">
          <string>Use 3D Delaunay Triangulation Search Space</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="Skip_self_intersection_check_box">
         <property name="text">
          <string>Skip Hole Fillings Making the Surface Self Intersecting</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Small Hole #Vertices Threshold</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="vertices_threshold_spin_box">
           <property name="decimals">
            <number>0</number>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>10.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Refinement Parameters</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Density Factor</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="Density_control_factor_spin_box">
           <property name="maximum">
            <double>96.989999999999995</double>
           </property>
           <property name="singleStep">
            <double>0.200000000000000</double>
           </property>
           <property name="value">
            <double>1.410000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Fairing Parameters</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Continuity</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="Continuity_spin_box">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>2</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Weight Type</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="weight_combo_box">
           <property name="currentIndex">
            <number>1</number>
           </property>
           <item>
            <property name="text">
             <string>Uniform Weight</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Cotangent Weight</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Hole Display and Selection</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="Visualize_holes_button">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Detect and Display Holes</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="Select_all_holes_button">
           <property name="text">
            <string>Select All Holes</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="Select_small_holes_button">
           <property name="text">
            <string>Select Small Holes</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="Deselect_all_holes_button">
           <property name="text">
            <string>Deselect All Holes</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Note: Use CTRL+Left mouse click to manually select/deselect a hole</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string>Actions</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="Fill_selected_holes_button">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Fill Selected Holes</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="Create_polyline_items_button">
           <property name="text">
            <string>Create Polyline Items for Selected Holes</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QPushButton" name="Reject_button">
        <property name="text">
         <string>Reject</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="Accept_button">
        <property name="text">
         <string>Accept</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
